src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -139,20 +138,45 @@ public function export(array $table)
             $query = $table['query'];
         }
 
-        try {
-            /** @var \PDOStatement $stmt */
-            $stmt = $this->executeQuery(
-                $query,
-                isset($table['retries']) ? (int) $table['retries'] : self::DEFAULT_MAX_TRIES
-            );
-        } catch (\Exception $e) {
-            throw $this->handleDbError($e, $table);
-        }
+        $maxTries = isset($table['retries']) ? (int) $table['retries'] : self::DEFAULT_MAX_TRIES;
```

Comment:
I don't get this. Why are you doing retry on `executeQuery` which is already implementing retry mechanism.

Reply:
Also these retry mechanisms here or in `executeQuery` looks unnecessarily complicated. For example why are you storing there that `$lastException`?

Reply:
It's remembering `lastException` so that it can log it in the retry message.